.github/workflows/cli-e2e-tests.yaml

```diff
@@ -36,29 +36,40 @@ jobs:
           AWS_DOCKER_ARTIFACT_REPO: ${{ secrets.AWS_DOCKER_ARTIFACT_REPO }}
           GIT_CREDENTIALS: ${{ secrets.GIT_CREDENTIALS }}
 
+      # Install Poetry.
+      - uses: snok/install-poetry@v1
```

Comment:
something like this can be in a composite action (e.g. https://github.com/aptos-labs/aptos-core/tree/main/.github/actions). You could then run it like `aptos-labs/aptos-core/.github/actions/python-setup@main` across any branch, repo, or workflow. Dependencies could be in the composite action as well, so we could avoid some inline dep installs

Reply:
So I'm not sure we need poetry for every Python installation. Though perhaps we actually do, I'd have to check.